lang/csharp/src/apache/main/Schema/UnionSchema.cs

```diff
@@ -71,11 +72,13 @@ internal static UnionSchema NewInstance(JArray jarr, PropertyMap props, SchemaNa
         /// Contructor for union schema
         /// </summary>
         /// <param name="schemas"></param>
-        /// <param name="props">dictionary that provides access to custom properties</param>
-        private UnionSchema(List<Schema> schemas, PropertyMap props) : base(Type.Union, props)
+        /// <param name="customProperties">dictionary that provides access to custom properties</param>
+        public UnionSchema(List<Schema> schemas, PropertyMap customProperties = null)
```

Comment:
Schema classes already use a Factory pattern method "NewInstance". I would recommend following the same pattern than exposing the constructor.